### ISSUE TITLE
Option parsing not correct in cli.py

### ISSUE THREAD

**Issue by wvalcke** [ultralytics/ultralytics#496]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

In the documentation about overriding the default config we have the following example:
yolo cfg=default_copy.yaml imgsz=320
This gives everyone the idea that you can give your own defaults in a yaml file, and overrule it by additional arguments. 
But this actually does not work, this principle only works when you overrule the default yaml (cfg=<file> argument not given).
The reason is that in cli.py in the function cli()
```
 if cfg.cfg:
        LOGGER.info(f"{PREFIX}Overriding default config with {cfg.cfg}")
        cfg = get_config(cfg.cfg)
```
The 'newer' default file is read in, but the existing overrides are not applied.
The solution is to give the overrides list also to cli() method and give it to get_config method which is foreseen to use an override list.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@wvalcke thanks for the bug report! This should be resolved in PR #506 scheduled for release 8.0.12 tomorrow.

**Comment by glenn-jocher** [comment]
@wvalcke this commit specifically should resolve the issue: https://github.com/ultralytics/ultralytics/pull/506/commits/48b37193c919fc6f4d249a9035b7a62de47f332f
